### PR TITLE
apcupsd - add another way to send email notifications

### PR DESCRIPTION
**Pull request by dbaio**
Add another way to send email notifications.
The user can choose between phpMailer or a simple SMTP mail client packaged with APCUpsd.
Bump version.

### CODE REVIEW

File: config/apcupsd/apcupsd.inc [hunk 1]
```diff
@@ -115,6 +115,28 @@ function validate_input_apcupsd($post,&$input_errors){
...
 
+		if ($post['emailnotification'] == 'smtpbuiltin') {
+
+			if (empty($post['smtphost'])) {
+					$input_errors[]='SMTP Host cannot be empty if SMTP builtin is enabled.';
+			}
+
+			if (empty($post['smtpport'])) {
+					$input_errors[]='SMTP Port cannot be empty if SMTP builtin is enabled.';
+			}elseif(!is_numericint($post['smtpport'])){
```
Comment (rbgarga) [inline config/apcupsd/apcupsd.inc]: There is a function called is_port() at util.inc, you should use it to validate port

File: config/apcupsd/apcupsd.inc [hunk 2]
```diff
@@ -207,10 +232,50 @@ function apccontrol_scripts_install(){
 		define('APCUPSD_BASE', '/usr/local');
 	}
 
+	$apcstatus[commfailure] = "\$HOSTNAME - Communications with UPS \$1 lost";
```
Comment (rbgarga) [inline config/apcupsd/apcupsd.inc]: Please use quotes for array key names: $apcstatus['commfailure']

File: config/apcupsd/apcupsd.inc [hunk 3]
```diff
@@ -207,10 +232,50 @@ function apccontrol_scripts_install(){
...
+					}else{
+						$smtpcc_cmd = "-c {$smtpcc}";
+					}
+				}
+			}
+
+			$apccontrol_script_file=<<<EOF
+#!/bin/sh
+
+HOSTNAME=`hostname`
```
Comment (rbgarga) [inline config/apcupsd/apcupsd.inc]: PHP has a function to get local hostname, it's called gethostname()

**Comment by dbaio** [comment]
thanks @rbgarga the code was changed and tested.

**Pull request merged**